File: drills/elixir/harnesses/nc25_min_window_substring.exs
[
  {~S{min_window("ADOBECODEBANC", "ABC")}, inspect("BANC"),
   inspect(Solution.min_window("ADOBECODEBANC", "ABC"))},
  {~S{min_window("a", "a")}, inspect("a"), inspect(Solution.min_window("a", "a"))},
  {~S{min_window("a", "aa")}, inspect(""), inspect(Solution.min_window("a", "aa"))},
  {~S{min_window("", "a")}, inspect(""), inspect(Solution.min_window("", "a"))},
  {~S{min_window("ab", "")}, inspect(""), inspect(Solution.min_window("ab", ""))},
  {~S{min_window("aaflslflsldkalskaaa", "aaa")}, inspect("aaa"),
   inspect(Solution.min_window("aaflslflsldkalskaaa", "aaa"))}
]

File: drills/ts/harnesses/nc25_min_window_substring.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.minWindow !== "function") throw new Error("__signature_mismatch__");
  return [
    ["minWindow('ADOBECODEBANC', 'ABC')", show("BANC"), show(solution.minWindow("ADOBECODEBANC", "ABC"))],
    ["minWindow('a', 'a')", show("a"), show(solution.minWindow("a", "a"))],
    ["minWindow('a', 'aa')", show(""), show(solution.minWindow("a", "aa"))],
    ["minWindow('', 'a')", show(""), show(solution.minWindow("", "a"))],
    ["minWindow('ab', '')", show(""), show(solution.minWindow("ab", ""))],
    ["minWindow('aaflslflsldkalskaaa', 'aaa')", show("aaa"), show(solution.minWindow("aaflslflsldkalskaaa", "aaa"))],
  ];
}
